packages/wallets/src/evm/connectors/token-bound-smart-wallet/index.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    AccountContractInfo,
+    FactoryContractInfo,
+    TokenBoundSmartWalletConfig,
+} from "./types"
+import { EVMWallet } from "../../interfaces";
+import * as module from "../smart-wallet";
+import { ethers } from "ethers";
+import { SmartContract } from "@thirdweb-dev/sdk";
+import { SmartWalletConfig } from "../smart-wallet/types";
+
+let SmartWalletConnector = module.SmartWalletConnector;
+
+export class TokenBoundConnnector extends SmartWalletConnector {
+    protected config: TokenBoundSmartWalletConfig;
+
+    constructor(config: TokenBoundSmartWalletConfig) {
+        const smartWalletConfig: SmartWalletConfig = {
+            chain: config.chain,
+            factoryAddress: config.factoryAddress,
+            thirdwebApiKey: config.thirdwebApiKey,
+            gasless: config.gasless,
+            bundlerUrl: config.bundlerUrl,
+            paymasterUrl: config.paymasterUrl,
+            paymasterAPI: config.paymasterAPI,
+            entryPointAddress: config.entryPointAddress,
+        }
+        super(smartWalletConfig);
+        this.config = config;
+    }
+
+    async initialize(personalWallet: EVMWallet) {
+        const factoryAddress = this.config.factoryAddress || "0x02101dfB77FDE026414827Fdc604ddAF224F0921";
+        await super.initialize(personalWallet, factoryAddress);
+    }
+
+    protected defaultAccountInfo(): AccountContractInfo {
+        return {
+            execute: async (account, target, value, data) => {
+                return account.prepare("executeCall", [target, value, data]);
+            },
+            getNonce: async (account) => {
+                return account.call("nonce", []);
```

Comment:
Dumb question but in the parent SmartWallet class, the calls here are: `execute` and `getNonce`, just making sure these are different on purpose

Reply:
Yeah in the official registry, the function signatures are different - they are of this format

Reply:
Can you double check that these can be configured back if they so choose 

@joaquim-verges could you check this please - my assumption was to use the erc-6551 registry function signatures by default rather than the ones we have - is this the route we are choosing to go? SHould we have a `custom` option or something which uses the thirdweb defaults? I just wanted to keep things simple but I'm not sure